src/unlimited.cpp

```diff
@@ -1153,7 +1154,23 @@ bool IsChainNearlySyncd()
     LOCK(cs_ischainnearlysyncd);
     return fIsChainNearlySyncd;
 }
-
+uint64_t LargestBlockSeen(uint64_t nBlockSize)
+{
+     // return the largest block size that we have seen since startup
+     static boost::atomic<uint64_t> nLargestBlockSeen;
+     static bool fInit = true;
+     if (fInit) {
+         nLargestBlockSeen.store(BLOCKSTREAM_CORE_MAX_BLOCK_SIZE);
+         fInit = false;
+     }
```

Comment:
This isn't thread safe. There is no ordering constraint on the flag, so it can be done in any order. You may end up seeing the variable uninitialized and the flag sset in another thread.